addons/controller_icons/Mapper.gd

```diff
@@ -32,8 +32,16 @@ func _convert_joypad_path(path: String, fallback) -> String:
 		_:
 			return ""
 
-func _get_joypad_type(fallback):
-	var controller_name = Input.get_joy_name(0)
+func _get_joypad_type(device, fallback):
+	var available = Input.get_connected_joypads()
+	if available.is_empty():
+		return fallback;
+	if !available.has(device):
+		device = ControllerIcons._last_controller
+	if !available.has(device):
+		device = available.front();
```

Comment:
- Add some comments here so it's clearer what these if tests are doing
- More idiomatic way to test element presence: `if not device in available:`
- Delete extra `;`
```suggestion
	# If the requested joypad is not on the connected joypad list, try using the last known connected joypad
	if not device in available:
		device = ControllerIcons._last_controller
	# If that fails too, then use whatever joypad we have connected right now
	if not device in available:
		device = available.front()
```

Reply:
Oh nice, I didn't know that `in` can be used this way in gdscript.